New package: ROOT v0.3.0

JuliaRegistrator opened the pull request:
- Registering package: ROOT
- Repository: https://github.com/JuliaHEP/ROOT.jl
- Created by: @grasph
- Version: v0.3.0
- Commit: 8094ac289cc962c3fe1240c70f2100acef935bdc
- Reviewed by: @grasph
- Reference: https://github.com/JuliaHEP/ROOT.jl/issues/22#issuecomment-2153472235
- Description: CERN ROOT compatiblility for Julia
<!-- bf0c69308befbd3ccf2cc956ac8a4671113b158fd87a6d4e741b0173dab051a48f7dc43e593ffff4357bc647a3a8359881cd8755c280d6680c6eeadb7c62f5dcc6ea7ee36d42b9ba34f30c84d13ee07aadbe0f7e5664a002594d5b670023b551a24cc6feb7ceb6e397f69dd4db568d7a58015d126c9f2c3894f03cd5126d09bf3023cb16899fbd6dc2cc67cc2bd69a08268daff2d5c20a740944c8e0ba5f706f -->

goerz commented:
Does this really need `conda` as a dependency?

goerz commented:
(Let me hold off on the "needs to be merged manually in 3 days" tag, since at least the compat-issues need to be fixed before it's ready to merge)

grasph commented:
The library it wraps is installed with Conda. Ultimately the BinaryProvider will be used, but the cross-compilation is a challenge for the ROOT library source, which is not yet solved yet (https://github.com/JuliaPackaging/Yggdrasil/pull/7666). Use of Conda was a  compromise in order to move on with the distribution of this wrapper.

For the compat specs what is the recommendations? Should I used major.minor of the latest version?

goerz commented:
> Should I used major.minor of the latest version?

That's what I usually do. Or, hypothetically, you could try to find out what are the oldest versions that actually work (make sure to have an entry in your test matrix that installs your dependencies in their declared oldest versions, to make sure it actually works and to tell you when you need to bump the compat entries).

JuliaRegistrator committed:
New package: ROOT v0.3.0

UUID: 1706fdcc-8426-44f1-a283-5be479e9517c
Repo: https://github.com/JuliaHEP/ROOT.jl.git
Tree: b9a6a9d56763bfc909a072539477666905fa10ae

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

grasph commented:
Can we proceed with the merging of this PR?

Let me know if I can do anything on my side. I don't understand the logs of AutoMerge: all the messages before the exception say "meets_this_guideline = true", but may be it is due to the package name, which requires a manual merge as discussed earlier.

Thanks,
Philippe.

goerz commented:
You’re okay, the only thing to do is to periodically ping the `#pkg-registration` channel on Slack. Usually things get merged quickly, but there’s been a backlog the past week. There’s a pretty small group of full maintainers who have merge permission, and maybe they’re all traveling/busy/on holiday, or something

grasph commented:
Thanks @goerz for this information. I've posted a message on the Slack channel.

grasph commented:
Thanks @ericphanson for the merge.
